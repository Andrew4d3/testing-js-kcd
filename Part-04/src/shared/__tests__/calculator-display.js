import React from "react";
import { render } from "@testing-library/react";
import CalculatorDisplay from "../calculator-display";

// (1) How can we easily assert outputs in Jest? Snapshots are the answer!
test("mounts", () => {
  const { container } = render(<CalculatorDisplay value="0" />);
  // (2) The first time we run this assertion method, a new snapshot will be created.
  // After a second time, Jest will compare the tested component with the snapshot created. If it's different, the test will fail and we will see the diff on the terminal
  expect(container.firstChild).toMatchSnapshot();
  // (3) If the changes on the snapshot are intended, we can update it by running jest with the -u flag or using the interactive terminal (watch mode)
});
