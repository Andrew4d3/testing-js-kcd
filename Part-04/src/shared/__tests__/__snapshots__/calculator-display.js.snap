// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`mounts 1`] = `
<div
  class="css-12149uf"
>
  <div
    class="autoScalingText"
    style="transform: scale(1,1);"
  >
    0
  </div>
</div>
`;
